tests/pytests/test_case.py

```diff
@@ -0,0 +1,375 @@
+from common import *
+
+def _prepare_index(env, idx, dim=4):
+    conn = getConnectionByEnv(env)
+    env.expect('FT.CREATE', idx, 'SCHEMA',
+               'v', 'VECTOR', 'FLAT', '6', 'DIM', dim, 'DISTANCE_METRIC', 'L2', 'TYPE', 'FLOAT32',
+               't', 'TEXT',
+               'x', 'NUMERIC').ok()
+
+    conn.execute_command('HSET', 'doc1', 'x', '1', 'v', create_np_array_typed([1.1] * dim).tobytes(), 't', 'pizza')
+    conn.execute_command('HSET', 'doc2', 'x', '1', 'v', create_np_array_typed([1.4] * dim).tobytes())
+    conn.execute_command('HSET', 'doc3', 'x', '1', 't', 'pizzas')
+    conn.execute_command('HSET', 'doc4', 'x', '1', 't', 'beer')
+    conn.execute_command('HSET', 'doc5', 'x', '1')
+
+def testWithVectorRange(env):
+    dim = 4
+    _prepare_index(env, 'idx', dim)
+    conn = getConnectionByEnv(env)
+
+    # Get vector distance and text score without APPLY
+    res = conn.execute_command(
+        'FT.AGGREGATE', 'idx', '@v:[VECTOR_RANGE .7 $vector]=>{$YIELD_DISTANCE_AS: vector_distance} | @t:(pizza) | @x==1',
+        'ADDSCORES', 'SCORER', 'BM25STD',
+        'PARAMS', '2', 'vector', create_np_array_typed([1.2] * dim).tobytes(),
+        'LOAD', '3', '@vector_distance', '@__key', 't',
+        'SORTBY', '2', '@__key', 'ASC',
+        'DIALECT', '2')
+    nresults = 5
+    env.assertEqual(res[0], nresults)
+    v_dist = [float(row[row.index('vector_distance') + 1] if 'vector_distance' in row else '0') for row in res[1:]]
+    t_score = [float(row[row.index('__score') + 1] if '__score' in row else '0') for row in res[1:]]
+
+    # Test APPLY with case function
```

Comment:
This is just prepare for test?
Isn't the test below after
```# Test APPLY with case function and linear combination of vector distance and text score```

Reply:
The section above obtains the vector distances and scores without `APPLY`.
The list `apply_expr_and_expected_results[]` contains the different expressions and expected results to test.
The `for`, execute the tests.